src/table_printer/table_printer.cpp

```diff
@@ -244,4 +273,96 @@ size_t table_printer::set_up_time_printing_interval(size_t tick) {
   return _tick_interval; 
 }
 
+#define _os_log_value_impl(format, column_index, value) \
+  os_log_info(                                          \
```

Comment:
Don't we need `#ifdef __APPLE__` here?

I'm assuming this needs to be a macro so that line numbers are preserved?

Also, could we centralize the preprocessor dependency on `__APPLE__` here by having the calls to `os_log_info` earlier in this file use this macro instead?

Reply:
Yes, we do need ifdef. I'll try to centralize the uses of ifdef at the top of the file.

Reply:
P.S. Yes, it does need to be a macro, and not just for line numbers to be preserved, but because there is a compile-time static assert that the format argument must be a string literal and not a char pointer (which could be dynamic).